tasks/structure.yml

```diff
@@ -0,0 +1,47 @@
+---
+- name: Ensure user
+  become: true
+  when: app_ensure_user
+  ansible.builtin.user:
+    name: "{{ app_user }}"
+    comment: "User for {{ app_name }} service"
+
+- name: Ensure group
+  become: true
+  when: "app_ensure_user and app_group != app_user"
+  ansible.builtin.user:
+    name: "{{ app_user }}"
+    groups:
+    - "{{ app_group }}"
+
+- name: Create the app directory
+  ansible.builtin.file:
+    path: "{{ app_root }}"
+    state: directory
+    mode: "0775"
+    owner: "{{ app_user }}"
+    group: "{{ app_group }}"
+
+- name: Create the data directory
+  ansible.builtin.file:
+    path: "{{ app_data_dir }}"
+    state: directory
+    mode: "0755"
+    owner: "{{ app_user }}"
+    group: "{{ app_group }}"
+
+- name: Create the app directory
+  ansible.builtin.file:
+    path: "{{ app_dir }}"
+    state: directory
+    mode: "0775"
+    owner: "{{ app_user }}"
+    group: "{{ app_group }}"
+
+- name: Create the config directory
+  ansible.builtin.file:
+    path: "{{ app_config_dir }}"
+    state: directory
+    mode: "0755"
```

Comment:
This should be `0750`.  Not all config is secret but it's better to assume it is.